src/movies.js

```diff
@@ -1,26 +1,73 @@
 // The `movies` array from the file `src/data.js`.
-console.log('movies: ', movies);
+//console.log('movies: ', movies);
+
+const movies = require("./data");
 
 
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(array) {
+  const directors = array.map((movie) =>  movie.director)
+  return directors;
+
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+function howManyMovies(array) {
+  const spielberg = array.filter(movie => {
+    return movie.director === 'Steven Spielberg' && movie.genre.includes('Drama');
+  })
+  return spielberg.length;
+}
+
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(array) {
+  if(!array.length){return 0;}
+  const scores = array.reduce(function(accumulator, currentValue){
+   if(!currentValue.score){
+    currentValue.score = 0 };
+    return accumulator + currentValue.score
+  },0);
+  const avg = scores /array.length
+    return Math.round(avg *100 + Number.EPSILON) / 100;
+
+  }
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore() {}
+function dramaMoviesScore(array) {
+  const dramas = array.reduce(function(accumulator, currentValue){
+    return accumulator + currentValue.genre.includes('Drama')},0);
+    return dramas;
+  const dramasAvg = dramas.reduce(function(accumulator, currentValue){
+    return accumulator + currentValue.score},0);
+    const dramaFilt = array.filter(function(movie){return movie.genre.includes('Drama')})
+}
```

Comment:
you can reuse the scoresAverage function instead of writing all the logic again and then it's just the following code for iteration 4:
```
 const dramaFilt = array.filter(function(movie){return movie.genre.includes('Drama')})

return scoresAverage(dramaFilt)
```